src/index.js

```diff
@@ -55,9 +74,31 @@ class ProfilerPlugin {
   }
 
   preInvoke() {
-    if (this.profilerEnabled) {
-      v8profiler.setSamplingInterval(this.config.sampleRate);
-      v8profiler.startProfiling(undefined, this.config.recSamples);
+    if (!this.enabled) {
+      return;
+    }
+
+    const promises = [];
+
+    try {
+      this.session.connect();
+
+      if (this.heapEnabled) {
+        promises.push(this.sessionPost('HeapProfiler.enable'));
+      }
+
+      if (this.profilerEnabled) {
+        promises.concat([
```

Comment:
This is definitely a style one, but why prefer `.concat` here over `.push` calls? I just had to mentally parse this a bit more with the `concat`

Reply:
Doing a single concat vs 3 push calls seems preferable to me.

Reply:
Yeah, I noticed it so I figured I'd float it by you. I'm good with it